app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -591,15 +599,42 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
         map!!.animateCamera(CameraUpdateFactory.newLatLngZoom(coordinates, zoomLevel))
     }
 
+    private fun toggleSaveButton() {
+        val saveButton : Button = requireActivity().findViewById(R.id.place_card_favorites_button)
+        if (saveButton.text.equals("Save")) {
+            saveButton.text = "Saved"
+            val redHeart : Drawable? = context?.resources?.getDrawable(R.drawable.ic_favorites_heart_red)
+            saveButton.setCompoundDrawablesWithIntrinsicBounds(redHeart, null, null, null)
+        } else {
+            saveButton.text = "Save"
+            val greyHeart : Drawable? = context?.resources?.getDrawable(R.drawable.ic_favorites_heart_grey)
```

Comment:
Avoid using depreciated method as much as possible 😄 

Please replace: 

`val greyHeart : Drawable? = context?.resources?.getDrawable(R.drawable.ic_favorites_heart_grey)`

with:

`val greyHeart = activity?.let { ContextCompat.getDrawable(it, R.drawable.ic_favorites_heart_grey) }`